docs/SMCWG-charter.md

```diff
@@ -0,0 +1,334 @@
+resiliency of the Internet shall be taken into consideration when the SMCWG
+forms its consensus. The SMCWG will consult other technical communities when and
+as necessary.
+
+The problem to be addressed by the working group is the absence of consistent
+and audited validation practices used by CAs in establishing the identity of the
+subject (if present) and verifying that the subscriber controls the email
+address. While there are methods for validating control of a domain that can be
+adopted from TLS certificates, there are currently no standard requirements for
+validating control of email addresses. Methods for validating identities in TLS
+certificates also exist and should be leveraged where possible, as well as other
+identity validation standards common in the industry.
+At least one method to efficiently validate an email address as well as
+developing a consistent profile for S/MIME certificates in order to facilitate
+technical interoperability across the Internet are also to be addressed by
+the SMCWG.
+
+## Chartering of the S/MIME Certificate Working Group
+
+A Chartered Working Group (“SMCWG”) is created to perform the activities as
+specified in this Charter, subject to the terms and conditions of the
+[CA/Browser Forum Bylaws](https://cabforum.org/bylaws/) and [Intellectual
+Property Rights (IPR) Policy](https://cabforum.org/ipr-policy/), as such
+documents may change from time to time. This charter for the S/MIME Certificate
+Working Group has been created according to CAB Forum Bylaw 5.3.1. In the event
+of a conflict between this Charter and any provision in either the Bylaws or
+the IPR Policy, the provision in the Bylaws or IPR Policy SHALL take
+precedence. The definitions found in the Forum’s Bylaws SHALL apply to
+capitalized terms in this Charter.
+
+### 1. Scope
+
+The authorized scope of the SMCWG SHALL be to discuss, adopt, and maintain
+policies, frameworks, and sets of standards related to the issuance and
+management of S/MIME certificates by third-party CAs under a publicly trusted
+root.
+
+The primary deliverable shall be scoped as follows:
+
+- Verification of control over email addresses
+- Key management and certificate lifecycle (subject to coordination with other
+  Forum CWGs to ensure consistency and avoid redundancy)
+- Certificate profiles for S/MIME certificates and Issuing CA certificates
+  (including the appropriateness of extensions and when those extensions should
+  be present)
+- CA operational practices, physical/logical security, etc.
+
+In addition to the primary deliverable, the SMCWG MAY also address:
+- Identity validation for natural persons and legal entities, in the context of
+  S/MIME certificates
+
+Deliverables of the SMCWG SHALL be constrained to those certificates
+that contain the emailProtection (OID: 1.3.6.1.5.5.7.3.4) or that are technically
+capable of such issuance.
+
+### 2. Out of Scope
+
+Certificates issued under a root certificate that is not publicly trusted SHALL
```

Comment:
It may be valuable to specify how such certificates are out of scope of any deliverables in a similar manner described in the servercert WG and code signing WG charters.

My proposed change, which borrows from the CSCWG charter:
"The SMCWG SHALL NOT develop guidelines, standards, or requirements applicable to certificates issued under a root certificate that is not publicly trusted".

Reply:
I’m not sure I see the connection between your problem statement and the proposed solution. Could you explain?

Your proposed change seems to restrict what the WG can work on, but in a way that itself isn’t defined and is, at best, seemingly self-referential to itself, since publicly trusted may ultimately be decided by what’s in scope.

Reply:
My intention is to better define how such untrusted certificates would be out of the scope of work for the WG, in a similar manner that the servercert WG and code signing WG charters exclude non-publicly trusted certificates from their deliverables. As the language currently stands, it's stated that such certificates are out of scope, but doesn't explain the net result/action of such exclusion (which, in this WG's case, would be not producing deliverables concerning them).

I drew inspiration from the language in 2 (c) of the CSWG charter as I thought that language fits the bill to both describe the certificates that are out of scope and how they will not be dealt with as part of the WG's deliverables.

Reply:
But doesn’t the current language solve that better than the proposed language?

Your proposal seems to prohibit the use of the work product, which the CABF can’t do, rather than describe the problem scope.

In order for the proposed language to work, “publicly trusted” would need to be defined here in a way that’s independent from the work product, but that also seems like an anti-goal?

I’m not trying to say the current language couldn’t be improved - it seems like it could - but merely for the problem I’m not sure I understand how the proposal is better than the current language, other than “This is what CSCWG did”. However, I’m probably misunderstanding and that’s why I wanted to try to figure the disconnect.

Reply:
I'm not sure how my proposed language would prohibit the use of such work products; the only prohibition is on the WG's development of such deliverables.

In other words, I don't think the proposed language would prevent some arbitrary non-public S/MIME PKI for adopting the WG deliverable as its CP. However, deliverables for such a PKI would be explicitly out of scope for the WG.

Reply:
But they wouldn’t, because “publicly trusted” is ill-defined here, and presumably end up self-referential (that is, “publicly trusted = use these standards”, which then defines any new requirement as functionally out of scope)

It seems there’s two dimensions here:
1. Clarifying that these requirements are voluntary and only apply to those who use them
2. Clarifying that the WG won’t impinge on other WGs (e.g. by focusing on EKU)

It seems we want both to be clear? Is that a fair statement?

Reply:
I think the definition for "publicly trusted" would be closer to "chains to a root certificate that is in the trust store of at least one Certificate Consumer", but I agree that it's better to use Defined Terms or similarly well-specified terms where possible to prevent differences in understanding.

As for the two dimensions, I think the second point is valuable to specify, although one could argue that it is already specified in section 1 ("Deliverables of the SMCWG SHALL be constrained to those certificates
that contain the emailProtection (OID: 1.3.6.1.5.5.7.3.4) or that are technically
capable of such issuance"). However, I'm not sure how the first point is relevant to defining the scope of work for a WG, as it seems to be speaking to the applicability of such deliverables to a given PKI.

Reply:
Yeah, I suggested "at least one Certificate Consumer" recently.  I also think "Root Certificate" is superior to "root certificate".  I'm not a huge fan of the current definition of "Root Certificate", but at least it exists.  "Publicly-Trusted Certificate" is also a defined term in the BRs, but it's (IMO) a bad definition, and shouldn't be used here.

Reply:
Does someone want to propose a change here?

Reply:
"Certificates issued under a Root Certificate that is not trusted by a Certificate Consumer SHALL be out of scope."

Reply:
@timfromdigicert

It would be great to avoid self-referential definitions. This seems especially important because the mere act of a single member (or one applying for membership) could change the entire scope of the WG. That's why I was trying to see if there was agreement in https://github.com/cabforum/documents/pull/167#discussion_r417736594 about what this language is trying to accomplish.

Put differently: Do we lose something of value by removing this statement entirely? It's not clear to me that we would.

Reply:
Yes, but neither of your options capture what this is trying to accomplish, which is important and shouldn't be removed.  As I stated on the public list, if we can't improve it, I support leaving it as is.

Reply:
@timfromdigicert If you believe there's something of value, could you rephrase what you believe is important here. Right now, this makes no sense as to what it's trying to accomplish, and I'd like to try to find a better way of rephrasing, especially if there's a misunderstanding about something that's important to you. I haven't seen any explanation from you, but let me know if I missed something.

Reply:
In my mind, the point is related to the following paragraph from the overview of the server BRs, parts of which may be worth stealing:

"This document describes an integrated set of technologies, protocols, identity-proofing, lifecycle management, and auditing requirements that are necessary (but not sufficient) for the issuance and management of Publicly-Trusted Certificates; Certificates that are trusted by virtue of the fact that their corresponding Root Certificate is distributed in widely-available application software. The requirements are not mandatory for Certification Authorities unless and until they become adopted and enforced by relying–party Application Software Suppliers."

I think it is important that the scope is explicitly limited to certificates that chain to Root Certificates that are trusted by Certificate Consumers who run root programs that voluntarily adopt the produced guidelines.

Reply:
Oh, I definitely think it's valuable context, but I'm not sure how it relates to what's out of scope.

What's an example of something that we want to use "Out of Scope" to prohibit? That might help figure out where the language is.

I can, for example, imagine someone saying "Oh, hey, we at Fido Corp use this policy, and we think it's useful". One argument is that even mentioning what Fido Corp does should be out of scope, because, see charter. But it's also in scope, because it's proposing that these certs should also consider what Fido Corp does.

Now, let's say Fido Corp wants to propose this, and we think it's not actually useful for these certs. The charter, as written, doesn't keep Fido Corp from bringing it up. All they have to do is join the Forum, and then suddenly their requirements on their certificates are relevant.

That's why I was suggesting "just delete", because everything I could think of that it would try to keep out, it wouldn't, and it would seem to keep out things it shouldn't (even though it doesn't)

Reply:
What is something it seems to keep out, but shouldn't?

Reply:
@timfromdigicert Did you perhaps miss https://github.com/cabforum/documents/pull/167#discussion_r421074233 which directly answered your question?

Reply:
I thought you had an explicit example beyond that.  Not only do I disagree that the language prohibits what you suggest, I can't see how an argument could be made that it does, so you'd have to walk me through why you believe that.

Reply:
@timfromdigicert 

I was asking you for an explicit example, to help find workable language. You can see it in the questions I've asked several times now:
https://github.com/cabforum/documents/pull/167#discussion_r421074233
> What's an example of something that we want to use "Out of Scope" to prohibit? That might help figure out where the language is.

https://github.com/cabforum/documents/pull/167#discussion_r420986318
> Put differently: Do we lose something of value by removing this statement entirely? It's not clear to me that we would.

I think it's useful, in order to make forward progress, if you could describe some important thing you feel that is worth keeping out, and that you're trying to do so with this language, so we can find better language.

https://github.com/cabforum/documents/pull/167#discussion_r421074233 is exactly the kind of example of how the language can be argued both ways, which you seem to be doing in https://github.com/cabforum/documents/pull/167#discussion_r424730355 if not aware, and that's why I want to try to find better language with you.

Reply:
To be clear, I'm fine with the current language.  As I have said several times, I think "root certificate" -> "Root Certificate" and defining public trust in terms of Certificate Issuers are useful improvements, but if there is not enough support for that, the current language is fine, and I believe no actual problems with it have been identified.  The burden is on those who want changes to propose and justify them.

Reply:
@timfromdigicert We're going in circles here.

I'm aware you're fine with the language. You have yet to describe what you want to accomplish with the language, which continues to be a problem to finding a productive solution to the problem that we are *not* fine with the language, and want to find a way to solve your problem without introducing new problems.

The attempt in https://github.com/cabforum/documents/pull/167#discussion_r421074233 was an attempt to understand your problem. You clearly believe it failed to do so, so again, I'm asking you to state an example of a thing you're trying to keep out of scope with this language.

Every single attempt to find something that is prohibited by this language I can find a way to make in-scope. The fact that anything can be read both ways *is a problem*, and that's the problem to solve: to make it clear and unambiguous what it is you want. The current language, unquestionably, does not do this.

Reply:
I think the out-of-scope language should be removed. The resulting CABF guideline should say that it is binding on those certificate consumers when they adopt it as their standard.

Reply:
Ryan, I haven't seen any objections to my email to the public list on Friday, so can you please delete this sentence?

Reply:
Done.